### PR TITLE
7996 Review NuGet package for Amido.Stacks.Core

### PR DESCRIPTION
**Pull request by alexanderwjrussell**
#### 📲 What

Review NuGet package for Amido.Stacks.Core.

#### 🤔 Why

Make all templates self contained.

#### 🛠 How

Replicating common files where required and ensuring they all implement the same pattern.

#### 👀 Evidence

No longer present in Shared:
image

All tests passing:
image

#### ✅ Acceptance criteria Checklist

- [ ] Code peer reviewed?
- [ ] Documentation has been updated to reflect the changes?
- [ ] Passing all automated tests, including a successful deployment?
- [ ] Passing any exploratory testing?
- [ ] Rebased/merged with latest changes from development and re-tested?
- [ ] Meeting the Coding Standards?

### CODE REVIEW

File: scripts/test-templates.sh [hunk 1]
```diff
@@ -51,6 +51,7 @@ dotnet new stacks-cqrs -n Cqrs.AllTheThings -do Menu --cloudProvider Azure --cic
 dotnet new stacks-cqrs -n Cqrs.ServiceBus -do Menu --cloudProvider Azure --cicdProvider AZDO -e ServiceBus
 dotnet new stacks-cqrs -n Cqrs.Dynamo -do Menu --cloudProvider AWS -db DynamoDb  
 dotnet new stacks-cqrs -n Cqrs.Sns -do Menu --cloudProvider AWS -e AwsSns
+dotnet new stacks-cqrs -n Cqrs.EventHub -do Menu --cloudProvider GCP --cicdProvider GHA -db DynamoDb -e EventHub
```
Comment (sharonrussell) [inline scripts/test-templates.sh:54]: We do not currently support GCP, is this line needed?

File: src/cqrs/src/api/xxENSONOxx.xxSTACKSxx.Common/Exceptions/ApplicationExceptionBase.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+public abstract class ApplicationExceptionBase : ApplicationException, IException, IOperationContext
+{
+    protected ApplicationExceptionBase(SerializationInfo info, StreamingContext context) : base(info, context) { }
+
+    public ApplicationExceptionBase() : base() { }
+
+    public ApplicationExceptionBase(string message) : base(message) { }
+
+    public ApplicationExceptionBase(string message, Exception ex) : base(message, ex) { }
+
```
Comment (sharonrussell) [inline src/cqrs/src/api/xxENSONOxx.xxSTACKSxx.Common/Exceptions/ApplicationExceptionBase.cs:16]: spacing, heathen

**Review by sharonrussell — CHANGES_REQUESTED** [review]
Just the one comment about the test-templates.sh. I'd either remove it or specify its purpose

File: scripts/test-templates.sh (hunk 1 above)
Comment (alexanderwjrussell) [reply scripts/test-templates.sh:54]: That is purely to generate the DevOps pipelines. The key purpose of this new template is EventHub. That was untested and therefore needed to be added.

**Pull request merged**